custom_model_runner/datarobot_drum/drum/gpu_predictors/nemo_predictor.py

```diff
@@ -94,6 +113,54 @@ def supported_payload_formats(self):
     def has_read_input_data_hook(self):
         return False
 
+    def _get_custom_artifacts(self):
+        code_dir_abspath = os.path.abspath(self._code_dir)
+
+        custom_py_paths = list(Path(code_dir_abspath).rglob("{}.py".format(CUSTOM_FILE_NAME)))
+        dot_remote_paths = list(Path(code_dir_abspath).rglob(DOT_REMOTE))
+
+        if len(custom_py_paths) + len(dot_remote_paths) > 1:
+            error_mes = (
+                "Multiple custom.py/.remote files were identified in the code directories sub directories.\n"
+                "The following custom model files were found:\n"
+            )
+            error_mes += "\n".join([str(path) for path in (custom_py_paths + dot_remote_paths)])
+            self.logger.error(error_mes)
+            raise DrumCommonException(error_mes)
+
+        return custom_py_paths, dot_remote_paths
+
+    def _load_model_remote_artifact(self):
+        from datarobot_custom_code.model_artifact_downloader import ModelDownloader
```

Comment:
@elatt
this import is somehow interferes with the NIM start. It downloads all artifacts successfully, but then NIM can't be started. If I start NIM with the uploaded model directly, it starts just fine.

Reply:
perhaps I need to deactivate environment properly, currently I just removing the `VIRTUAL_ENV/bin:` from the `PATH`.

Reply:
Yeah, my first guess would be with venv (I haven't looked in a long time what all the magic `deactivate` shell function does). You can probably just update your subprocess call to `shell=True` and then do `deactivate && <cmd to run NIM>` right?

Reply:
Actually there isn't that much magic
```sh
deactivate () {
    # reset old environment variables
    if [ -n "${_OLD_VIRTUAL_PATH:-}" ] ; then
        PATH="${_OLD_VIRTUAL_PATH:-}"
        export PATH
        unset _OLD_VIRTUAL_PATH
    fi
    if [ -n "${_OLD_VIRTUAL_PYTHONHOME:-}" ] ; then
        PYTHONHOME="${_OLD_VIRTUAL_PYTHONHOME:-}"
        export PYTHONHOME
        unset _OLD_VIRTUAL_PYTHONHOME
    fi

    # Call hash to forget past commands. Without forgetting
    # past commands the $PATH changes we made may not be respected
    hash -r 2> /dev/null

    if [ -n "${_OLD_VIRTUAL_PS1:-}" ] ; then
        PS1="${_OLD_VIRTUAL_PS1:-}"
        export PS1
        unset _OLD_VIRTUAL_PS1
    fi

    unset VIRTUAL_ENV
    unset VIRTUAL_ENV_PROMPT
    if [ ! "${1:-}" = "nondestructive" ] ; then
    # Self destruct!
        unset -f deactivate
    fi
}
```
- Cleanup `PATH`
- Cleanup `PYTHONHOME` (only relevant if base image had PYTHONHOME set for some reason) 
- cleanup `PS1` (not relevant for your script)
- cleanup some other venv related env vars (not sure but also probably not relevant)
- run `hash` (pretty sure not relevant when we run a command via `subprocess` with `shell=False`)

So, in summary, I think as long as you are fixing up `PATH` correctly, I think things should work.